tests/filesystem/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+#![no_std]
+#![feature(const_str_as_bytes)]
+#![feature(const_raw_ptr_deref)]
+
+extern crate alloc;
+
+use linux_kernel_module::filesystem::{self, FileSystem, FileSystemFlags};
+use linux_kernel_module::{self, CStr};
+
+struct TestFSModule {
+    _fs_registration: filesystem::Registration<TestFS>,
+}
+
+// This is safe because Registration doesn't actually expose any methods.
+unsafe impl Sync for TestFSModule {}
+
+struct TestFS {}
+
+impl FileSystem for TestFS {
+    const NAME: &'static CStr = unsafe { &*("testfs\x00" as *const str as *const CStr) };
```

Comment:
Ooof, is `cstr!()` not `const`? That seems like the correct fix!

Reply:
Thanks for the review!

> That seems like the correct fix!

Do you mean that we should create a `const_cstr!` or that what I did is ok? I tried around a bit and we could postpone the check performed in `CStr::new` when the string is dereferenced (then it could be created at compile-time). However this would be inefficient and also would break the rule stated in the [core::ops::Deref docs](http://doc.rust-lang.org/1.37.0/core/ops/trait.Deref.html) that `deref` should never fail.

We could also create a macro that ensures that `\x00` is appended but does not check for the length to match the position of the 0-byte. I've got very little experience with Rust, so if you have any preference can you maybe give me a hint on what might be more _rusty_?

Reply:
`cstr!()` itself should just be `const` I think.

I think your proposal, to have the macro append the `\x00`, but not do the check that there are no other `0`s in it, is the correct one.